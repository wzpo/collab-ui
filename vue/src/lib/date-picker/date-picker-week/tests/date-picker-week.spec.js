import { shallowMount } from '@vue/test-utils';
import DatePickerWeek from '../index.vue';
import DatePickerDay from '../../date-picker-day/index.vue';
import Button from '../../../button/index.vue';
import Vue from 'vue';
import moment from 'moment-timezone';

Vue.component(Button.name, Button);
Vue.component(DatePickerDay.name, DatePickerDay);

const TEST_DATE = '2018-04-01';

describe('DatePickerWeek', () => {
  let day;

  beforeAll(() => {
    // setting timezone to CST
    moment.tz.setDefault('Asia/Kolkata');
    day = moment(TEST_DATE);
  });

  afterAll(() => {
    moment.tz.setDefault();
  });

  it('should match snapshot', () => {
    const container = shallowMount(DatePickerWeek, {
      propsData: {
        day: day,
        month: 3,
      },
    });
    expect(container.html()).toMatchSnapshot();
  });

  it('should render a div', () => {
    const container = shallowMount(DatePickerWeek, {
      propsData: {
        day: day,
        month: 3,
      },
    });
    expect(container.findAll('.md-datepicker__week').length).toEqual(1);
  });

  it('should render 7 Day components', () => {
    const container = shallowMount(DatePickerWeek, {
      propsData: {
        day: day,
        month: 3,
      },
    });
    expect(container.findAll(DatePickerDay)).toHaveLength(7);
  });
});

