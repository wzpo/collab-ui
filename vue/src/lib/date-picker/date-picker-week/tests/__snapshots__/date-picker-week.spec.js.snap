// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePickerWeek should match snapshot 1`] = `
<div class="md-datepicker__week">
  <md-date-picker-day-stub day="Sun Apr 01 2018 00:00:00 GMT+0530" month="3"></md-date-picker-day-stub>
  <md-date-picker-day-stub day="Mon Apr 02 2018 00:00:00 GMT+0530" month="3"></md-date-picker-day-stub>
  <md-date-picker-day-stub day="Tue Apr 03 2018 00:00:00 GMT+0530" month="3"></md-date-picker-day-stub>
  <md-date-picker-day-stub day="Wed Apr 04 2018 00:00:00 GMT+0530" month="3"></md-date-picker-day-stub>
  <md-date-picker-day-stub day="Thu Apr 05 2018 00:00:00 GMT+0530" month="3"></md-date-picker-day-stub>
  <md-date-picker-day-stub day="Fri Apr 06 2018 00:00:00 GMT+0530" month="3"></md-date-picker-day-stub>
  <md-date-picker-day-stub day="Sat Apr 07 2018 00:00:00 GMT+0530" month="3"></md-date-picker-day-stub>
</div>
`;
